client-v2/src/shared/util/snap.ts

```diff
@@ -14,34 +16,32 @@ function validateId(id: string) {
   );
 }
 
-function convertToSnap({ id, ...rest }: ArticleFragment): ArticleFragment {
-  let href;
-
-  if (isGuardianUrl(id)) {
-    href = '/' + getAbsolutePath(id, true);
-  } else {
-    href = id;
-  }
-
-  return {
+function convertToSnap({
+  id,
+  ...rest
+}: PartialBy<ArticleFragment, 'id'>): ArticleFragment {
+  const fragment = {
     id: generateId(),
     ...rest,
-    meta: {
-      href,
-      ...rest.meta
-    }
+    meta: rest.meta
   };
+
+  if (!id) {
+    return fragment;
+  }
+
+  const href = isGuardianUrl(id) ? '/' + getAbsolutePath(id, true) : id;
+  return set(['meta', 'href'], href, fragment);
 }
 
 async function createLinkSnap(
-  url: string,
+  url?: string,
   meta?: ArticleFragmentMeta
 ): Promise<ArticleFragment> {
   const uuid = v4();
   try {
-    const { title, description, siteName } = meta
-      ? ({} as any)
-      : await fetchOpenGraphData(url);
+    const { title, description, siteName } =
+      meta || !url ? ({} as any) : await fetchOpenGraphData(url);
```

Comment:
Can you add the new line back in here, just to make clear this is a ternary. Bit easier to follow

Reply:
Sadly, prettier is in charge here :(

Reply:
ah that is a shame. will it let you away with `(meta || !url)` at least?